### PR TITLE
 Add size, height and width for crop layer and switch order layer.

### LINKED ISSUE #5887: Add size, height and width for crop layer and switch order layer.

### PR DESCRIPTION
**Pull request by wanghaoshuang**
fix #5887

### CODE REVIEW

File: python/paddle/trainer/config_parser.py [hunk 1]
```diff
@@ -3850,6 +3859,16 @@ def __init__(self, name, inputs, reshape, **xargs):
...
         self.config.reshape_conf.width_axis.extend(reshape['width'])
+        input_layer = self.get_input_layer(0)
+        if reshape is None:
+            self.set_layer_size(input_layer.size)
+        else:
+            inH = input_layer.height
+            inW = input_layer.width
+            inC = input_layer.size / inH / inW
+            out_dims = [0, inH, inW, inC]
+            size = reduce(lambda x, y: x * y, out_dims[reshape['width'][0]:])
```
Comment (chengduoZH) [inline python/paddle/trainer/config_parser.py]: The input data for SwitchOrderLayer may be five dimensions (NCDHW).

Comment (wanghaoshuang) [reply python/paddle/trainer/config_parser.py]: Thx.

Comment (qingqing01) [reply python/paddle/trainer/config_parser.py]: naming format: inH -> in_h, inW->in_w, ..

Comment (wanghaoshuang) [reply python/paddle/trainer/config_parser.py]: Fixed.

**Pull request merged**